### PR TITLE
rpidistro-ffmpeg: fix opengl compile issue

### PR DESCRIPTION
**Pull request by EasyIP2023**
Connected to: https://github.com/agherzan/meta-raspberrypi/issues/1117

**Problem:**
rpidistro-ffmpeg fails the `./configure` stage if neither x11 or wayland defined
in `DISTRO_FEATURES`. After configure stage succeeds `compile` stage
fails as `SelectedGetProcAddress` isn't defined. One can't define it as
GLES2/gl2.h `glGetProcAddress` if not defined. Anywhere actually.

Also couldn't find the original locations of where I got the patches from, but if
you go to

https://github.com/RPi-Distro/ffmpeg/tree/ea72093f350f38edcd39c480b331c3219c377642/debian/patches

One can see that the patches applied were incorrect.

EDIT:

found the [ffmpeg-4.3.2-rpi_10.patch](https://github.com/RPi-Distro/ffmpeg/blob/378adea166f167a0e75cf85888f661101cb4af0f/debian/patches/ffmpeg-4.3.2-rpi_10.patch)

https://github.com/RPi-Distro/ffmpeg/tree/378adea166f167a0e75cf85888f661101cb4af0f

**Solution:**

Add two new patches so that `./configure` is OE ported and
`libavdevice/opengl_enc.c compiles` if `SelectedGetProcAddress` isn't
defined. It can't however be define as `glGetProcAddress` is not defined
anywhere. Patch applied loads GL functions pointers at compile time versus
dynamically at runtime.

Also MR was checked with all `./configure` flags. Only one to fail was gsm due to libgsm not found.

### CODE REVIEW

File: recipes-multimedia/rpidistro-ffmpeg/rpidistro-ffmpeg_4.3.2.bb [hunk 1]
```diff
@@ -33,23 +33,31 @@ RPROVIDES:${PN} = "${PROVIDES}"
 DEPENDS = "nasm-native"
 
 inherit autotools pkgconfig
-PACKAGECONFIG ??= "avdevice avfilter avcodec avformat swresample swscale postproc avresample \
-                   opengl udev sdl2 ffplay alsa bzlib lzma pic pthreads shared theora zlib \
-                   libvorbis x264 gpl sand rpi vout-drm vout-egl \
+PACKAGECONFIG ??= "avdevice avfilter avcodec avformat swresample swscale postproc avresample ffplay \
```
Comment (agherzan) [inline recipes-multimedia/rpidistro-ffmpeg/rpidistro-ffmpeg_4.3.2.bb:36]: A fair bit of PACKAGECONFIG changes around. We will need to either split them in a separate commit or mention/explain them in git log.

Comment (EasyIP2023) [reply recipes-multimedia/rpidistro-ffmpeg/rpidistro-ffmpeg_4.3.2.bb:36]: Yes, closing this one. The next one should hopefully have a better explanation.
